### PR TITLE
Create log storage adapter service

### PR DESCRIPTION
**Pull request by hasYuting**
Create the taffyDB storage adapter service
Manage the service dependencies between the improved $log service and the taffyDB service

### CODE REVIEW

File: fn.logger.js [hunk 1]
```diff
@@ -7,7 +7,7 @@
         $provide.decorator("$sce", [ "$delegate", function($delegate) {
             return $sce = $delegate;
         } ]);
-        $provide.decorator("$log", [ "$delegate", function($delegate) {
+        $provide.decorator("$log", [ "$delegate", "TaffyDBAdapter", function($delegate, TaffyDBAdapter) {
```
Comment (jackofseattle) [inline fn.logger.js]: There should be no default storage adapter most likely.  This still requires taffy to be installed to work, that's no fun.

Comment (hasdavidc) [inline fn.logger.js]: how would you "choose" the adapter to be used?

Comment (hasYuting) [inline fn.logger.js]: So I should inject it into the directive not in config stage?

Comment (jackofseattle) [inline fn.logger.js]: Provider options.

we decorate log here which is cool, but we should allow the user to configure some sort of logDB provider

**Pull request closed without merging**